rust/connlib/shared/src/control.rs

```diff
@@ -341,6 +319,54 @@ pub struct PhoenixMessage<T, R> {
     reference: Option<String>,
 }
 
+#[derive(Debug, PartialEq, Eq, Deserialize, Serialize, Clone)]
+#[serde(tag = "event", content = "payload")]
+enum Payload<T, R> {
+    #[serde(rename = "phx_reply")]
+    Reply(Reply<R>),
+    #[serde(rename = "phx_error")]
+    Error(Empty),
+    #[serde(rename = "phx_close")]
+    Close(Empty),
+    #[serde(rename = "disconnect")]
+    Disconnect { reason: String },
+    #[serde(untagged)]
+    Message(T),
+}
```

Comment:
@conectado I used the revise the implementation of `Payload`. Flattening it out turns out to be a much simpler implementation IMO. I also opted to name all the variants explicitly so it easier to follow what is going on here. It also makes them "greppable" :)